client/layout/guided-tours/index.js

```diff
@@ -23,22 +24,48 @@ class GuidedTours extends Component {
 		return this.props.tourState !== nextProps.tourState;
 	}
 
-	next = ( tour, nextStepName ) => {
+	next = ( { tour, tourVersion, nextStepName, skipping = false } ) => {
+		// Don't immediately record the transition to the next step. Instead,
+		// remember the name of the step and _maybe_ record it later (at the
+		// next invokation of `next` or `quit`), depending on whether the step
+		// was skipped (see Step#skipIfInvalidContext).
```

Comment:
In this case wouldn't the comment be worded better if it said something along the lines:

> Before advancing to the next step in the tour, record the transition for the current step if it was not skipped.